diff --git a/jsPsych_subjective_listening_tests_GES/js/debug_experiment.js b/jsPsych_subjective_listening_tests_GES/js/debug_experiment.js
new file mode 100644
index 0000000..6be6595
--- /dev/null
+++ b/jsPsych_subjective_listening_tests_GES/js/debug_experiment.js
@@ -0,0 +1,18 @@
+var all_trials = [
+    // Practice sessions
+    {
+        "subset": 0,
+        "session": 0,
+        "file": "data/backups/male/174-84280-0006_3536-23268-0011_01/174-84280-0006_3536-23268-0011_01_4kHz.wav",
+        "scale": "SIG",
+        "end_experiment": "false"
+    },
+    // Test sessions:
+    {
+        "subset": 1,
+        "session": 0,
+        "file": "data/male_speakers/5694-64025-0003_1673-143397-0011_01/5694-64025-0003_1673-143397-0011_01_500Hz.wav",
+        "scale": "SIG",
+        "end_experiment": "false"
+    },
+]
diff --git a/jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.json b/jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.js
similarity index 99%
rename from jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.json
rename to jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.js
index e25e354..3cde62d 100644
--- a/jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.json
+++ b/jsPsych_subjective_listening_tests_GES/js/subjective_listening_experiment_1.js
@@ -1,4 +1,4 @@
-all_trials = [
+var all_trials = [
     // Practice sessions
     {
         "subset": 0,
@@ -589,4 +589,4 @@ all_trials = [
         "scale": "OVRL",
         "end_experiment": "false"
     }
-]
\ No newline at end of file
+]
diff --git a/jsPsych_subjective_listening_tests_GES/js/submit.js b/jsPsych_subjective_listening_tests_GES/js/submit.js
new file mode 100644
index 0000000..00d2edc
--- /dev/null
+++ b/jsPsych_subjective_listening_tests_GES/js/submit.js
@@ -0,0 +1,28 @@
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function submit(json_data, code) {
+    for(let i=0; i<3; i++) {
+        const myHeaders = new Headers();
+        myHeaders.append("Content-Type", "application/json");
+    
+        try {
+            const response = await fetch("/submit/" + code, {
+                method: "POST",
+                body: JSON.stringify(json_data),
+                headers: myHeaders,
+            });
+    
+            if (!response.ok) {
+                // Wait a bit and re-try
+                console.log('error submitting results, waiting then trying again')
+                await sleep(1500);
+            } else {
+                return; // All done
+            }
+        } catch (error) {
+            console.log('submit error: ' + error)
+        }
+    }
+}
diff --git a/jsPsych_subjective_listening_tests_GES/subjective_test_1.html b/jsPsych_subjective_listening_tests_GES/subjective_test_1.html
index f530c89..c4496a2 100644
--- a/jsPsych_subjective_listening_tests_GES/subjective_test_1.html
+++ b/jsPsych_subjective_listening_tests_GES/subjective_test_1.html
@@ -26,7 +26,9 @@
 
   <!-- Load experiment configuration file -->
   <script src="js/config.js" type="text/javascript"></script> 
-  <script src="js/subjective_listening_experiment_1.json" type="text/javascript"></script> 
+  <!--<script src="js/subjective_listening_experiment_1.js" type="text/javascript"></script> -->
+  <script src="js/debug_experiment.js" type="text/javascript"></script> 
+  <script src="js/submit.js" type="text/javascript"></script> 
 
   <!-- Load CSS styles -->
   <link href="../lib/jspsych-7.3.0/dist/jspsych.css" rel="stylesheet" type="text/css"></link>
@@ -60,7 +62,9 @@ console.log(all_trials.length)
 var jsPsych = initJsPsych({
       on_finish: function() {
       //jsPsych.data.displayData(); 
-        jsPsych.data.get().localSave('csv',`${uniqueID}_data.csv`);
+        //jsPsych.data.get().localSave('csv',`${uniqueID}_data.csv`);
+        var data = jsPsych.data.get().json();
+        submit(data, 'random');
        },
       on_trial_start: function() {
        },
